operator/src/test/java/oracle/kubernetes/operator/MainTest.java

```diff
@@ -475,6 +475,19 @@ void withLabelSelector_onCreateReadNamespaces_startsNamespaces() {
     assertThat(getStartingNamespaces(), contains(NS_WEBLOGIC1, NS_WEBLOGIC3, NS_WEBLOGIC5));
   }
 
+  @Test
+  void withLabelSelector_onCreateReadNamespaces_ignoreSelectorOnList_startsNamespaces() {
```

Comment:
This test name is unclear. What is it trying to verify?

I'm very concerned about the `ignoreSelectorOnListOperation` call. What K8s behavior is being simulated or tracked?

Reply:
The behavior/condition the change is trying to simulate is when the readExistingNamespace async call is issued, the strategy is List, so the results is as if the selector does not exist although the strategy is LabelSelector when the returned valued are processed.

Reply:
I'm not following that; if the strategy has been changed to LabelSelector, why is K8s acting as though it is List? Are we sending the wrong call? Do we have an in-flight list call going when the strategy is changed? If so, maybe we should address that in the list response?

Reply:
yes, the in-flight call uses List. We simulate this race condition by ignoring the selector.

Reply:
In the product code, we do address the issue in the list response as the changes in DomainRecheck.

Reply:
As to the test method name, what about withLabelSelector_returnAllNamespacesOnCreateReadNamespaces_startsExpectedNamespaces?

Reply:
> Having thought more about the problem, I think we should not have used the label selector in the list call for LabelSelector case. We should always list all namespaces (without a selector), and filter the returned list with the selector. All other strategies are handled this way.

That's fine, and lends itself to a simple solution: NamespaceListResponseStep.onSuccess() can call a new method, `getMetadatas` rather `getNames` and the Namespaces.isDomainNamespace() method can take a @NonNull V1ObjectMeta rather than a strong.

Reply:
As to the test, I would suggest that it should now test the change of strategy between the list call and the response. In that case, the thing to add to KubernetesTestSupport would be a `doOnList` call, similar to `doOnCreate`, etc. The test would then set the strategy to, say, List and have the `doOnList` change it to LabelSelection. That could be shown to fail without any other changes.

I would name such a test, `whenSelectionStrategyChangesDuringList_startDesiredNamespaces`.

As a rule, it is better to have the test name describe the desired behavior rather than the implementation of the test.

Reply:
we don't need the new test case any more. One of the existing test case covers it already now that we changed the listNamespaces for LabelSelector case.